clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -55,12 +58,17 @@
 
   public static synchronized TachyonFileSystem get() {
     if (sTachyonFileSystem == null) {
-      sTachyonFileSystem = new TachyonFileSystem();
+      boolean enableLineage = ClientContext.getConf().getBoolean(Constants.USER_ENABLE_LINEAGE);
+      if (enableLineage) {
+        return TachyonLineageFileSystem.get();
+      } else {
+        sTachyonFileSystem = new TachyonFileSystem();
+      }
```

Comment:
Can you elaborate it more? A factory class inside this class? That wouldn't make a much difference, right?